cdap-ui/app/cdap/components/RulesEngineHome/RuleBookDetails/RulebookMenu/AddRulesEngineToPipelineModal.js

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+/*
+  1. Get batch and realtime artifacts.
+  2. Check if yare is present
+  3. Make a batch and realtime config with yare in its stage.
+  4. Show the modal.
+*/
+import React, {Component, PropTypes} from 'react';
+import {MyArtifactApi} from 'api/artifact';
+import NamespaceStore from 'services/NamespaceStore';
+import {findHighestVersion} from 'services/VersionRange/VersionUtilities';
+import {Modal, ModalHeader, ModalBody} from 'reactstrap';
+import MyRulesEngine from 'api/rulesengine';
+import classnames from 'classnames';
+import T from 'i18n-react';
+
+require('./AddRulesEngineToPipelineModal.scss');
+const PREFIX = 'features.RulesEngine.AddRulesEngineToPipelineModal';
+
+export default class AddRulesEngineToPipelineModal extends Component {
+  static propTypes = {
+    rulebookid: PropTypes.string,
+    isOpen: PropTypes.bool,
+    onClose: PropTypes.func
+  };
+
+  state = {
+    rulebookid: this.props.rulebookid,
+    isOpen: false,
+    batchUrl: '',
+    realtimeUrl: '',
+    batchConfig: null,
+    realtimeConfig: null
+  };
+
+  updateBatchAndRealtimeConfigs = () => {
+    let {selectedNamespace: namespace} = NamespaceStore.getState();
+    MyArtifactApi
+      .list({namespace})
+      .combineLatest([
+        MyRulesEngine
+          .getRulebook({
+            namespace,
+            rulebookid: this.props.rulebookid
+          })
+      ])
+      .subscribe(
+        (res) => {
+          let artifacts = res[0];
+          let rulebook = res[1].values[0];
+          let batchArtifacts = artifacts.filter(artifact => artifact.name === 'cdap-data-pipeline');
+          let realtimeArtifacts = artifacts.filter(artifact => artifact.name === 'cdap-data-streams');
+          let yareArtifact = artifacts.filter(artifact => artifact.name === 'yare-plugins');
+          let batchHighestArtifact, realtimeHighestArtifact;
+          if (batchArtifacts.length > 1) {
+            batchHighestArtifact = findHighestVersion(batchArtifacts.map(artifact => artifact.version), true);
```

Comment:
I don't think this returns the highest artifact object, i think this just returns the highest version string or version object